src/hotspot/cpu/x86/macroAssembler_x86.cpp

```diff
@@ -7963,6 +7963,77 @@ void MacroAssembler::cache_wbsync(bool is_pre)
     sfence();
   }
 }
+
+void MacroAssembler::copy64_masked_avx(Register dst, Register src, XMMRegister xmm,
+                                       KRegister mask, Register length, Register temp,
+                                       BasicType type, int offset, bool use64byteVector) {
+  assert(MaxVectorSize >= 32, "vector length < 32");
```

Comment:
Why does "MaxVectorSize >= 32" imply that "vector length < 32"?

This assert appears in multiple locations.